hudi-hadoop-common/src/main/java/org/apache/hudi/io/hadoop/HoodieAvroOrcReader.java

```diff
@@ -86,9 +88,16 @@ public ClosableIterator<IndexedRecord> getIndexedRecordIterator(Schema readerSch
 
     Configuration hadoopConf = storage.getConf().unwrapAs(Configuration.class);
     try (Reader reader = OrcFile.createReader(new Path(path.toUri()), OrcFile.readerOptions(hadoopConf))) {
-      TypeDescription orcSchema = AvroOrcUtils.createOrcSchema(readerSchema);
-      RecordReader recordReader = reader.rows(new Options(hadoopConf).schema(orcSchema));
-      return new OrcReaderIterator<>(recordReader, readerSchema, orcSchema);
+      TypeDescription fileSchema = reader.getSchema();
```

Comment:
why cann't we just read the file with reader schema? The colunar file projection is more efficient than the avro rewrite.

Reply:
It was for the schema evolution. I was able to push this down into the Avro conversion logic